### DISCUSSION TITLE
Add difficulty filtering to multi

### DISCUSSION THREAD

**Discussion by toppasaurus** [ppy/osu#31241] [Ideas]
Sometimes, you get trolls who drop 6* maps into 2* rooms, which is annoying. I've also seen cases where higher-skill players basically chased the novice players out by getting a bit over-excited and dropping a bunch of 4-5*+ maps, which is somewhat problematic. The second  is uncommon, but I foresee it becoming more common as the player base moves to Lazer.

I suggest having a button in the room settings to enable the host to restrict map difficulty much the same way as it's done in map select. It could take the form of matching the filter in the host's map select, or being a static setting editable by the host similar to the other room settings. The difficulty range could then be displayed in the lobby.

This basically turns the honor code system of putting a difficulty range in the room title into concrete enforcement.

**Comment by WalmartWhiteCat** [comment]
iirc this is 100% solved with an upcoming update (I don't know when, I just heard it in passing). It'll allow people to choose which difficulty they want to play on the chosen mapset. So, say I put on something like Flamewall (The 10* one), but someone else doesn't want to mess around playing a map that's too hard for them. That player would be able to go into the mapset and choose to play any other available difficulty in the set, while not affecting the difficulty that any others in the lobby will be playing.

But don't quote me on this. I'm kind of just parroting some things I've heard others say and vaguely recalling some things I read. I don't actually have 100% clarity on if this update will happen or not or if it's even possible under the current multiplayer system.

**Reply by toppasaurus** [reply]
The issue here is that there are a significant number of mapsets which don't fit in certain difficulty ranges. I often vibe in rooms that have ranges like 3.7-4.5* or whatever. I have many single-diff maps which are 4.9+\*, and a number of mapsets with no diff above 2.5*, which don't fit that. Add the diff that fits to prove it exists, then play whatever you want.

Edited because markdown is jank sometimes

**Comment by Xeladoes** [comment]
https://github.com/ppy/osu/discussions/32032
I shared my more detailed opinion there, I also hope lazer can add filter with better star rating algorithm.

Thank you for supporting normal players!
